pkg/storage/internal/actions/internalize.go

```diff
@@ -38,26 +40,38 @@ func newInternalizeAction(
 	knownTxRepo KnownTxRepo,
 	outputRepo OutputRepo,
 	random wdk.Randomizer,
+	chaintracker chaintracker.ChainTracker,
 ) *internalize {
 	logger = logging.Child(logger, "internalizeAction")
 	return &internalize{
-		logger:      logger,
-		txRepo:      txRepo,
-		basketRepo:  basketRepo,
-		knownTxRepo: knownTxRepo,
-		outputRepo:  outputRepo,
-		random:      random,
+		logger:       logger,
+		txRepo:       txRepo,
+		basketRepo:   basketRepo,
+		knownTxRepo:  knownTxRepo,
+		outputRepo:   outputRepo,
+		random:       random,
+		chaintracker: chaintracker,
 	}
 }
 
 func (in *internalize) Internalize(ctx context.Context, userID int, args *wdk.InternalizeActionArgs) (*wdk.InternalizeActionResult, error) {
-	tx, err := transaction.NewTransactionFromBEEF(args.Tx)
+	beef, txIDHash, err := transaction.NewBeefFromAtomicBytes(args.Tx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create transaction from BEEF: %w", err)
+		return nil, fmt.Errorf("failed to create atomic beef from bytes: %w", err)
 	}
-	txID := tx.TxID().String()
 
-	// TODO: Do SPV verification of the transaction - it requires Services::ChainTracker
+	if ok, err := beef.Verify(ctx, in.chaintracker, false); err != nil {
+		return nil, fmt.Errorf("failed to verify beef: %w", err)
+	} else if !ok {
+		return nil, fmt.Errorf("beef verification returned false")
+	}
+
+	tx := beef.FindAtomicTransactionByHash(txIDHash)
+	if tx == nil {
+		return nil, fmt.Errorf("problem with atomic beef: transaction with hash %s not found", txIDHash)
+	}
```

Comment:
Some suggestion to make it slightly more readable and some ideas for error wording changes, what do you think?
```suggestion
	ok, err := beef.Verify(ctx, in.chaintracker, false)
        if err != nil {
	        return nil, fmt.Errorf("failed to verify beef: %w", err)
        }
        if !ok {
	        return nil, fmt.Errorf("beef verification failed")
        }

        tx := beef.FindAtomicTransactionByHash(txIDHash)
        if tx == nil {
	        return nil, fmt.Errorf("atomic beef error: transaction with hash %s not found")
        }
```

Reply:
I rephrased the errors - including your suggestions.
But, regarding ifs and `err/ok` variables, see below comment 👇